warp/service.go

```diff
@@ -23,22 +24,18 @@ var errNoValidators = errors.New("cannot aggregate signatures from subnet with n
 
 // API introduces snowman specific functionality to the evm
 type API struct {
-	networkID                     uint32
-	sourceSubnetID, sourceChainID ids.ID
-	backend                       Backend
-	state                         validators.State
-	client                        peer.NetworkClient
-	requirePrimaryNetworkSigners  func() bool
+	chainContext                 *snow.Context
+	backend                      Backend
+	signatureGetter              aggregator.SignatureGetter
+	requirePrimaryNetworkSigners func() bool
 }
 
-func NewAPI(networkID uint32, sourceSubnetID ids.ID, sourceChainID ids.ID, state validators.State, backend Backend, client peer.NetworkClient, requirePrimaryNetworkSigners func() bool) *API {
+func NewAPI(chainCtx *snow.Context, networkCodec codec.Manager, backend Backend, client peer.NetworkClient, requirePrimaryNetworkSigners func() bool) *API {
+	signatureGetter := aggregator.NewSignatureGetter(client, networkCodec)
```

Comment:
optional: we could take the signatureGetter as an arg instead of client + networkCodec, it seems

also note that we would want to apply these changes to subnet-evm but we can do that at the end (when merging the base branch)